### PR TITLE
Add DaemonSet conditions

### PR DESCRIPTION
**Pull request by janetkuo**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
xref https://github.com/kubernetes/features/issues/353, #7856

**Special notes for your reviewer**: 
A `conditions` field is added to DaemonSetStatus for consistency. No valid DaemonSet condition type is supported yet. 
@kubernetes/sig-apps-api-reviews 

**Release note**:

```release-note
DaemonSet status now has a new field named "conditions", making it consistent with other workloads controllers.
```

### CODE REVIEW

File: staging/src/k8s.io/api/apps/v1beta2/types.go [hunk 1]
```diff
@@ -601,6 +601,36 @@ type DaemonSetStatus struct {
...
+
+// DaemonSetCondition describes the state of a DaemonSet at a certain point.
+type DaemonSetCondition struct {
+	// Type of DaemonSet condition.
+	Type DaemonSetConditionType `json:"type" protobuf:"bytes,1,opt,name=type,casttype=DaemonSetConditionType"`
+	// Status of the condition, one of True, False, Unknown.
+	Status v1.ConditionStatus `json:"status" protobuf:"bytes,2,opt,name=status,casttype=k8s.io/api/core/v1.ConditionStatus"`
+	// The last time this condition was updated.
+	// +optional
+	LastUpdateTime metav1.Time `json:"lastUpdateTime,omitempty" protobuf:"bytes,3,opt,name=lastUpdateTime"`
```
Comment (smarterclayton) [inline staging/src/k8s.io/api/apps/v1beta2/types.go]: Same question as for stateful sets - should we omit this field since we believe it's use is limited?

Comment (janetkuo) [reply staging/src/k8s.io/api/apps/v1beta2/types.go]: Agree. We can omit it for now.

**Comment by janetkuo** [comment]
Removed `LastUpdateTime` field from `DaemonSetCondition`. Deployment will be the only workload API with `LastUpdateTime` in its Condition.

**Comment by dixudx** [comment]
@janetkuo pull-kubernetes-e2e-kops-aws flake #55303

**Pull request merged**